rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,657 @@
+        removed:
+          type: array
+          items:
+            type: string
+          description: List of properties that were removed (and not updated)
+        notPresent:
+          type: array
+          items:
+            type: string
+            description:
+              List of properties requested for removal that were not found
+              in the namespace's properties.
+              Represents a partial success response.
+              Server's do not need to implement this.
+    ListNamespacesResponse:
+      type: object
+      properties:
+        namespaces:
+          type: array
+          items:
+            $ref: '#/components/schemas/Namespace'
+    DropTableResponse:
+      type: object
+      properties:
+        dropped:
+          type: boolean
+          description: true if the table was found and removed from the catalog
+        purged:
+          type: boolean
+          description: whether the underlying data was purged or is being purged
+    Namespace:
+      type: array
+      description: Reference to one or more levels of a namespace
+      items:
+        type: string
+      example: [ "accounting", "tax" ]
+    GetNamespaceResponse:
+      type: object
+      properties:
+        namespace:
+          $ref: '#/components/schemas/Namespace'
+        properties:
+          type: object
+          example: { "owner": "Ralph", 'transient_lastDdlTime': '1452120468' }
+    ListTablesResponse:
+      type: object
+      properties:
+        identifiers:
+          type: array
+          items:
+            $ref: '#/components/schemas/TableIdentifier'
+    IcebergConfiguration:
+      type: object
+      description: "Server-provided configuration for the catalog."
+      properties:
+        rootPath:
+          minLength: 1
+          type: string
+          default: /
+          nullable: false
+          description:
+            Root path to be used for all other requests. Server-side implementations
+            are free to use another choice for root path, but must conform to the
+            specification otherwise in order to interoperate with the URI generation
+            that the REST catalog client will do.
+        catalogProperties:
+          type: object
+          nullable: true
+          description:
+            An optional field for storing catalog configuration properties that
+            are stored server side. This could be beneifical to an administrator,
+            for example to enforce that a given LocationProvider is used or to
+            enforce that a certain FileIO implementation is used.
+
+    TableAlreadyExistsError:
+      allOf:
+        - $ref: '#/components/schemas/ResponseErrorObject'
+      description: the given table identifier already exists and cannot be created / renamed to
+
+  #############################
+  # Reusable Response Objects #
+  #############################
+  responses:
+
+    IcebergResponseObject:
```

Comment:
Wrapping a successful response into another `data` field is not something I have seen many APIs doing. Convention seems to be that the true object to be at the top level? Is the motivation for that choice to be able to quickly identify if the response is an error message from a successful one?

Reply:
Yeah that's the motivation.

Admittedly in some languages that's useful, in other's not so much. It also provides standardized form for the first top level of the message.

But you are right typically most APIs just send their response, as it's already a JSON object. I'd be fine to remove this, but it might also make the OpenAPI document definition a bit cleaner (though I'm open to any suggestions about using the OpenAPI document I might not be aware of if you have a lot of experience with using them).

Reply:
In my own experience, having the actual body as the top level object is not really an issue in Java/Javascript. It should also simplify a bit your OpenAPI definitions? Currently ResponseDataObjects is defined as an object, but for each operation it would be better to have something more defined, and I'm not sure how easy it is to add extra constraints on the `data` field